rasterio/transform.py

```diff
@@ -224,8 +225,13 @@ def rowcol(transform, xs, ys, op=math.floor, precision=None):
     if not isinstance(ys, Iterable):
         ys = [ys]
 
-    eps = sys.float_info.epsilon
-    
+    if precision is None:
+        eps = sys.float_info.epsilon * (1.0 - 2.0 * op(0.1))
```

Comment:
PR #2106 missed the sign change depending on the value of `op`, which shifts up or down properly.

Reply:
I think this (https://github.com/mapbox/rasterio/blob/8ff3305e4adc1b87b5ca9e6ed0f03bd28a1c069f/rasterio/windows.py#L293) would break the logic here.

Reply:
If the purpose of op is to change sign if it "rounds up", maybe we could use a more explicit check.
```python
if op(0.1) == 1:
    eps = -eps
```

Reply:
Thank you, that's exactly the kind of feedback I was looking for :smiley: